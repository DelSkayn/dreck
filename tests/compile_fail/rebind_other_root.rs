use dreck::*;

fn main() {
    new_root!(owner1, root1);
    new_root!(owner2, root2);

    let ptr = root1.add(0);

    let ptr = unsafe{ rebind(ptr) };
    let ptr = root2.rebind_to(ptr);

    root1.collect_full(owner1);
    assert_eq!(*ptr.borrow(owner2),0);
}
