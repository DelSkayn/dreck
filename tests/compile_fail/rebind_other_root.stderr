error[E0597]: `tag` does not live long enough
  --> tests/compile_fail/rebind_other_root.rs:5:5
   |
5  |     new_root!(owner2, root2);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
...
14 | }
   | -
   | |
   | `tag` dropped here while still borrowed
   | borrow might be used here, when `_cell_owner` is dropped and runs the `Drop` code for type `main::new_cell_owner`
   |
   = note: values in a scope are dropped in the opposite order they are defined
   = note: this error originates in the macro `new_root` (in Nightly builds, run with -Z macro-backtrace for more info)
